src/snapred/backend/recipe/algorithm/GroupDiffractionCalibration.py

```diff
@@ -106,6 +112,21 @@ def unbagGroceries(self):
         self.wsTOF: str = self.getPropertyValue("InputWorkspace")
         self.focusWS: str = str(self.getPropertyValue("GroupingWorkspace"))
 
+        # create string names of workspaces that will be used by algorithm
+        self.outputWStof: str = ""
+        if self.getProperty("OutputWorkspace").isDefault:
+            self.outputWStof = wng.diffCalOutput().runNumber(self.runNumber).build()
+            self.setProperty("OutputWorkspace", self.outputWStof)
+        else:
+            self.outputWStof = self.getPropertyValue("OutputWorkspace")
+
+        self.maskWS: str = ""
+        if self.getProperty("MaskWorkspace").isDefault:
+            self.maskWS = wng.diffCalMask().runNumber(self.runNumber).build()
+            self.setProperty("MaskWorkspace", self.maskWS)
+        else:
+            self.maskWS = self.getPropertyValue("MaskWorkspace")
+
```

Comment:
I think I lean in favor of just making these fields required.  They arent user facing, and the dependence on the input data to determine a good default name seems to indicate they should be supplied.

Reply:
I concur, they should simply be required.  See my comment above on how to do that.

Reply:
Please see my comment below.  These don't necessarily have to be `Input` parameters, they could also be optional (or mandatory) `Output` parameters.